narwhals/dependencies.py

```diff
@@ -176,6 +176,45 @@ def is_pandas_like_series(arr: Any) -> bool:
     return is_pandas_series(arr) or is_modin_series(arr) or is_cudf_series(arr)
 
 
+def is_into_series(ser: Any) -> bool:
+    """
+    Check whether `ser` can be converted to a Narwhals Series.
+
+    Arguments:
+        ser: The object to check.
+
+    Returns:
+        `True` if `ser` can be converted to a Narwhals Series, `False` otherwise.
+
+    Examples:
+        >>> import pandas as pd
+        >>> import polars as pl
+        >>> import numpy as np
+        >>> import narwhals as nw
+
+        >>> s_pd = pd.Series([1, 2, 3])
+        >>> s_pl = pl.Series([1, 2, 3])
+        >>> np_arr = np.array([1, 2, 3])
+
+        >>> nw.dependencies.is_into_series(s_pd)
+        True
+        >>> nw.dependencies.is_into_series(s_pl)
+        True
+        >>> nw.dependencies.is_into_series(np_arr)
+        False
+    """
+
+    return (
+        (
+            hasattr(ser, "_compliant_series")
+            and hasattr(ser._compliant_series, "__narwhals_series__")
+        )
```

Comment:
Would this translate to:

```suggestion
    from narwhals.series import Series
    
    return (
        isinstance(ser, Series)
```

or is there any implication with the stable API?

Reply:
I think we want both, so something like
```
if isinstance(ser, Series)
or hasattr(ser, '__narwhals_series__')
or is_polars_series(ser)
# etc.
```

We're trying to find objects which, if we pass them to `from_native` with `allow_series=True` or `series_only=True`, then they will successfully get converted to a Narwhals Series

And in `from_native`, these are the things we check for

Reply:
Thanks. Will take a look and make the required updates.

Reply:
I've been looking at the checks in `from_native`.  Just so I make sure I completely understand - is `hasattr(ser, '__narwhals_series__')` to account for passing in a compliant series (number 3 below).

So, you'll get `True` for:

1. a polars, pandas series etc:
```
is_into_series(pl.Series([1, 2, 3]))
```
2. for `nw` series:
```
nw.from_native(pd.Series([1, 2, 3]), allow_series=True)
```
3. And for the underlying `_compliant_series`?
```
nw.from_native(pd.Series([1, 2, 3]), allow_series=True)._compliant_series
```

Reply:
thanks for asking, definitely good to get clarity on these things!

My hope is that Narwhals can just be glue between compliant implemetations. Currently, all compliant implementations live in the Narwhals codebase itself (`narwhals/_pandas_like`, `narwhals/_polars`, ...), by my hope is that the [extensibility mechanism](https://narwhals-dev.github.io/narwhals/extending/#extending-narwhals) can allow other dataframe libraries to define their own Narwhals-compliant implementation, so that all they expose to us is `__narwhals_dataframe__` / `__narwhals_series__` / `__native_namespace__` (and some others).

Some reasons why people might want to do this (as opposed to having the wrapping code live in Narwhals) are:
- they might want to be in control of the implementation and only hand off to us the bare minimum
- they might want the Narwhals-compatible code to be written in a lower-level language (C, Rust, ...), whereas anything which lives in Narwhals itself must be pure-Python
- they might be a closed-source library which wants to expose the bare minimum

Granted, nobody has yet made such an extension (for a start, the docs don't make it very easy to understand exactly what needs to be done...). But I'm hoping to be able to focus more on it next year, and to provide a reference implementation

Please do ask if this isn't clear, thanks 🙏

Reply:
This makes sense to me now. Thanks so much for the detail explanation! 
Really interested to learn more about what it would look like to create a Narwhals-compliant implementation and would love to help in some way with the documentation on the topic.